### PR TITLE
Improve code quality

### PR DESCRIPTION
**Pull request by JoelHo**
Add testing
Resolve TODOs

### CODE REVIEW

File: src/test/java/seedu/address/storage/JsonAdaptedPoolTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+
+    @Test
+    public void toModelType_invalidTags_throwsIllegalValueException() {
+        List<JsonAdaptedTag> invalidTags = new ArrayList<>(VALID_TAGS);
+        invalidTags.add(new JsonAdaptedTag(INVALID_TAG));
+        JsonAdaptedPool pool =
+                new JsonAdaptedPool(VALID_TRIPDAY, VALID_TRIPTIME, VALID_PASSENGERS, invalidTags, VALID_DRIVER);
+        assertThrows(IllegalValueException.class, pool::toModelType);
+    }
```
Comment (chewterence) [inline src/test/java/seedu/address/storage/JsonAdaptedPoolTest.java:85]: i guesss this is okay, but some of these tests could be refactored out and put under one umbrella test method that tests invalid tags. Within the test they are seperated by comments // that tells the reader what the tests are

File: src/test/java/seedu/address/testutil/TypicalAddressBook.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import seedu.address.model.person.passenger.Passenger;
+import seedu.address.model.pool.Pool;
+
+public class TypicalAddressBook {
+    private TypicalAddressBook() {} // prevents instantiation
+
+    /**
+     * Returns an {@code AddressBook} with all the typical pools and passengers.
+     */
+    public static AddressBook getTypicalAddressBook() {
```
Comment (chewterence) [inline src/test/java/seedu/address/testutil/TypicalAddressBook.java]: perhaps could write full java docs with the @params

File: src/test/java/seedu/address/storage/JsonAdaptedPoolTest.java (hunk 1 above)
Comment (JoelHo) [reply src/test/java/seedu/address/storage/JsonAdaptedPoolTest.java:85]: The point of the tests is to test the whole chain when reading a JsonAdaptedPool, I don't think there is any issue here.

File: src/test/java/seedu/address/testutil/TypicalAddressBook.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * A utility class containing an {@code AddressBook} object to be used in tests.
+ */
+public class TypicalAddressBook {
+    private TypicalAddressBook() {} // prevents instantiation
+
+    /**
+     * Returns an {@code AddressBook} with all the typical pools and passengers.
+     * @return An {@code AddressBook} with all the typical pools and passengers.
```
Comment (chewterence) [inline src/test/java/seedu/address/testutil/TypicalAddressBook.java]: java docs should have a space between the description and @return

**Review by lyueyang — APPROVED** [review]
LGTM from quality perspective

**Review by ZechariahTan — APPROVED** [review]
LGTM

**Pull request merged**